docs/collections.md

```diff
@@ -126,61 +109,57 @@ function formatDate(date) {
-+   const routes = allTags.map((tag) => ({ tag }));                          // this is where we set { params: { tag } }
-
-    return {
--     async data() {
--       return allPosts;
-+     async data({ params }) {
-+       return allPosts.filter((post) => post.tags.includes(params.tag));    // filter posts that match the :tag from the URL ("params")
-      },
-      pageSize: 10,
-+     routes,
-+     permalink: ({ params }) => `/tag/${params.tag}/`                       // this is where we generate our URL structure
-    };
-  }
-  ---
-```
-
-Some important concepts here:
+## Example: Grouping Content by Tag, Author, etc.
 
-- `routes = allTags.map((tag) => ({ tag }))`: Astro handles pagination for you automatically. But when it needs to generate multiple routes, this is where you tell Astro about all the possible routes. This way, when you run `astro build`, your static build isn’t missing any pages.
-- `permalink: ({ params }) => `/tag/${params.tag}/`: this is where you tell Astro what the generated URL should be. Note that while you have control over this, the root of this must match the filename (it’s best **NOT** to use `/pages/$tag.astro`to generate`/year/$year.astro`; that should live at `/pages/$year.astro` as a separate file).
-- `allPosts.filter((post) => post.tag === params.tag)`: we aren’t returning all posts here; we’re only returning posts with a matching tag. _What tag,_ you ask? The `routes` array has `[{ tag: 'javascript' }, { tag: '…`, and all the routes we need to gather. So we first need to query everything, but only return the `.filter()`ed posts at the very end.
-
-Other things of note is that we are sorting like before, but we filter by the frontmatter `tag` property, and return those at URLs.
+```jsx
+---
+// Define the `collection` prop.
+export let collection: any;
 
-These are still paginated, too! But since there are other conditions applied, they live at a different URL.
+// Define a `createCollection` function. 
+// In this example, we'll customize the URLs that we generate to
+// create a new page to group every pokemon by first letter of their name. 
+export async function createCollection() {
+  const allPokemonResponse = await fetch(`https://pokeapi.co/api/v2/pokemon?limit=150`);
+  const allPokemonResult = await allPokemonResponse.json();
+  const allPokemon = allPokemonResult.result;
+  const allLetters = ['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z'];
+  return {
+    // `routes` defines the total collection of routes. 
+    // A route is represented by a `params` data object (instead of a string).
+    // In this example, we format each letter (ex: "a") to params (ex: {letter: "a"}).
+    routes: allLetters.map(letter => {
+      const params = {letter};
+      return params;
+    }),
+    // `permalink` defines the final URL for each route object defined in `routes`.
+    permalink: ({ params }) => `/pokemon/${params.letter}`,
```

Comment:
Not sure if it fits, but it could be that the user intends to change the URL /permalink even if not grouping data like this. So this could become a subsection.

Reply:
I'm not sure if we really want to encourage this (vs. placing the collection file where you want it) since it breaks with file-based routing a bit

Reply:
Maybe I'll just add that as a note to the docs on this line